can_tools/scrapers/official/AZ/AZ_TableauPull.py

```diff
@@ -0,0 +1,280 @@
+import requests
+import urllib.parse
+from bs4 import BeautifulSoup
+from can_tools.scrapers.base import CMU
+from can_tools.scrapers.official.base import StateDashboard
+import pandas as pd
+import json
+import re
+
+
+class ArizonaData(StateDashboard):
+    """
+    Fetch county level covid data from Arizona's Tableau dashboard
+    """
+
+    # Initialize
+    source = "https://www.azdhs.gov/preparedness/epidemiology-disease-control/infectious-disease-epidemiology/covid-19/dashboards/index.php"
+    has_location = True
+    location_type = "county"
+    state_fips = 4
+    cntys = [
+        ["APACHE", 4001],
+        ["COCHISE", 4003],
+        ["COCONINO", 4005],
+        ["GILA", 4007],
+        ["GRAHAM", 4009],
+        ["GREENLEE", 4011],
+        ["LA PAZ", 4012],
+        ["MARICOPA", 4013],
+        ["MOHAVE", 4015],
+        ["NAVAJO", 4017],
+        ["PIMA", 4019],
+        ["PINAL", 4021],
+        ["SANTA CRUZ", 4023],
+        ["YAVAPAI", 4025],
+        ["YUMA", 4027],
+    ]
+    # Short list of request parameters for state-level db
+    reqParams = {":embed": "y", ":display_count": "no"}
+
+    def fetch(self):
+        outDf = pd.DataFrame()
+        # Extracts the map filter function, used to build tableau subview URLs
+        def getMapFilter(htmDump):
+            urlFltr = []
+            # Grab the map filter function guts:
+            for fn in htmDump["worldUpdate"]["applicationPresModel"][
+                "workbookPresModel"
+            ]["dashboardPresModel"]["userActions"]:
+                if fn.get("name") == "Map filter":
+                    urlFltr = (
+                        urllib.parse.unquote(fn.get("linkSpec").get("url"))
+                        .split("?")[1]
+                        .replace("=<Countynm1~na>", "")
+                        .split("&")
+                    )
+            return urlFltr
+
+        def extractData(htmdump, area):  # Extracts data from raw tableau HTML bootstrap
```

Comment:
I think it might generalize, but like we saw with California data, different states are using Tableau differently, and that will impact how we form a parent class. 

From what I've seen, we almost want a few different Tableau parent classes:

    -  Mapclick Subview Data Extraction (Arizona-like):  Pull the onclick map subview query function.  Then use the filter expression it to generate per-county subviews. Then scrape data from these subviews.
    
    -  External Inpage Filter (California-like):  The query expression is not linked to a mapclick, but externally on the page. We need to find that filter expression, pipe it to Tableau to generate county-specific views (these are not subviews), and then scrape the data from there.